dvc/repo/ls.py

```diff
@@ -76,9 +76,9 @@ def onerror(exc):
             if not recursive:
                 for dname in dirs:
                     info = PathInfo(root) / dname
-                    if not dvc_only or (
-                        tree.dvctree and tree.dvctree.exists(info)
-                    ):
+                    # pylint:disable=protected-access
+                    _, dvctree = tree._get_tree_pairs(info)  # noqa
```

Comment:
Temporary for now, maybe we won't need it after `tree.stat()` impl.

Reply:
So maybe let's start with `tree.stat()`?

Reply:
This is a pre-requisite for `tree.stat()`, as I would need to change things again to account for subrepos, and it's not that intrusive for now.